.github/workflows/e2e/k8s/e2e-chart/templates/sample-job.yml

```diff
@@ -38,6 +38,10 @@ spec:
           value: "tracecontext,baggage"
         - name: OTEL_GO_AUTO_INCLUDE_DB_STATEMENT
           value: "true"
+        - name: OTEL_BSP_SCHEDULE_DELAY
+          value: "60000"
+        - name: OTEL_BSP_MAX_EXPORT_BATCH_SIZE
+          value: {{ .Values.expectedSpansCount | quote }}
```

Comment:
I thought the plan was to leave this value alone? Setting this value only reduces it so far. If the point of this change is to fix the flakiness, leaving it where it is will avoid changes like aecea372c91677d968e1b7b4bab90d211d813748

Reply:
I do think that this will reduce the flakiness. 
The number of spans in a test is not something I expect to be changed often.
The configuration in this PR as I understand it allows to specify number of spans in a test and if we don't get that amount after a minute it will fail.

Reply:
But the test doesn't terminate when a batch is exported, it stops when the job completes. The spans are flushed regardless of this setting. If we export 2 spans but the is set to 3 then the job will stop, the SDK will flush the batch processor, and then the evaluation will run as normal.

If we want to test missing spans we would need to do that in the bats validation.

This change will only affect things when we send more spans than the configured value. It will split things into two batches.

Reply:
I tried the approach we talked about. It seems we are not calling `tracerProvider.Shutdown`. I tried to add this during the manager closing phase and it helped in exporting the spans.
I am still not 100% sure about this not adding flakiness, since we are also using a file exporter in the e2e-test to create the JSON file, so even if the BSP is flushed there still may be a chance of the file not being ready?
@damemi

Reply:
@RonFed you could try configuring the batch exporter in the collector's config https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/main/.github/workflows/e2e/k8s/collector-helm-values.yml

afaik the Go SDK should automatically flush when it receives a sigterm like we're sending, @MrAlias is that right? I don't think a manual call to Shutdown is necessary in that case

Reply:
> afaik the Go SDK should automatically flush when it receives a sigterm like we're sending, @MrAlias is that right? I don't think a manual call to Shutdown is necessary in that case

No, an explicit call to `Shutdown` or `ForceFlush` is necessary. There is no signal capture in the SDK.

FWIW, if we weren't calling `Shutdown` on the SDK when we were shutting down, we should definitely change that.

Reply:
@MrAlias ahh, that was an assumption I had when writing the e2es

Reply:
@MrAlias is going to open a PR to add a shutdown to the opentelemetry.Controller and make sure that flushes the TracerProvicer.

That should unblock this.

Reply:
https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1028